.github/workflows/style.yml

```diff
@@ -0,0 +1,42 @@
+name: style
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  check_whitespace:
```

Comment:
As the two steps are pretty much identical, you could make that a matrix strategy build with `check: [whitespace, permissioons]` and uses `cmake --build build --target check-${{ matrix.check }}` below.